libs/platform/src/lib/components/input-group/input-group.component.ts

```diff
@@ -0,0 +1,159 @@
+import {
+    Component,
+    Input,
+    ChangeDetectionStrategy,
+    ContentChildren,
+    QueryList,
+    ElementRef,
+    ViewEncapsulation,
+    ChangeDetectorRef,
+    Renderer2,
+    AfterViewInit,
+    OnInit,
+    AfterContentInit
+} from '@angular/core';
+import { startWith } from 'rxjs/operators';
+
+import { BaseComponent } from '../base';
+import { InputComponent } from '../form/input/input.component';
+import { ContentDensity, Status } from '../form/form-control';
+
+import { InputGroupConfig } from './input-group.config';
+import { InputGroupAddonComponent } from './addon/addon.component';
+
+const CSS_CLASS_NAME = {
+    inputGroup: 'fdp-input-group',
+    inputGroupInnerInput: 'fd-input-group__input'
+} as const;
+
+/**
+ * Fundamental input group component
+ *
+ * ```html
+ * <fdp-input-group>
+ *   <fdp-input-group-addon>$</fdp-input-group-addon>
+ *   <fdp-input type="number" name="price"></fdp-input>
+ * </fdp-input-group>
+ * ```
+ *
+ */
+@Component({
+    selector: 'fdp-input-group',
+    templateUrl: './input-group.component.html',
+    styleUrls: ['./input-group.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InputGroupComponent extends BaseComponent implements OnInit, AfterContentInit, AfterViewInit {
```

Comment:
If I am thinking about usecase, where you need to instantiate InputGroup on the fly (with ComponentResorverFactory & ViewContainerRef), how would you initialize it? 

Or for each specific usecase you would have a wrapper that uses input group? 

MySalaryComponent, that would e.g. use this component and add $ as prefix ?

Reply:
So usually we deal with a template and in this case we have to write something like that:
```
<fdp-input-group name="salary">
    <fdp-input-group-addon>$</fdp-input-group-addon>
    <fdp-input-group-input type="number"></fdp-input-group-input>
</fdp-input-group>
```
But if we want to create it dynamically it seems to be a problem since angular allows put projected content only as a string or a html node as far as I know.
```
viewContainerRef.createComponent(componentFactory, 0, null, [
  [
    document.createTextNode('First ng-content'), 
    document.createElement('br'), 
    document.createTextNode('First ng-content second row')
  ]
]);
```

Reply:
To create it dynamically we have to use input-group core implementation, because core version does not use projections.
But core version is limited to have only one addon while this approach allows to have several addons.